File: 007/reverse-integer.go
package main

import "fmt"

func reverse(x int) int {
	var (
		y    int
		re   = 0
		maxn = 2147483647
	)
	if x < 0 {
		y = -x
	} else {
		y = x
	}
	for y > 0 {
		if re != 0 && maxn/re < 10 {
			return 0
		}
		re *= 10
		re += y % 10
		y /= 10
	}
	if x < 0 {
		return -re
	}
	return re
}

func main() {
	x := 123
	fmt.Println(reverse(x))
}

File: 007/reverse-integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        y = -x if x < 0 else x
        re = 0
        maxn = 2147483647
        while y > 0:
            if re != 0 and maxn / re < 10:
                return 0
            re *= 10
            re += int(y % 10)
            y = int(y / 10)

        return -re if x < 0 else re
